client/genericclient/stream.go

```diff
@@ -0,0 +1,220 @@
+		return serviceinfo.GenericMethod
+	}
+}
+
+type clientStreamingClient struct {
+	streaming.Stream
+	svcInfo *serviceinfo.ServiceInfo
+	method  string
+}
+
+func NewClientStreaming(ctx context.Context, genericCli Client, method string, callOpts ...callopt.Option) (ClientStreaming, error) {
+	gCli, ok := genericCli.(*genericServiceClient)
+	if !ok {
+		return nil, errors.New("invalid generic client")
+	}
+	stream, err := getStream(ctx, genericCli, method, callOpts...)
+	if err != nil {
+		return nil, err
+	}
+	return &clientStreamingClient{stream, gCli.svcInfo, method}, nil
+}
+
+func (cs *clientStreamingClient) Send(req interface{}) error {
+	_args := cs.svcInfo.MethodInfo(cs.method).NewArgs().(*generic.Args)
+	_args.Method = cs.method
+	_args.Request = req
+	return cs.Stream.SendMsg(_args)
+}
+
+func (cs *clientStreamingClient) CloseAndRecv() (resp interface{}, err error) {
+	if err := cs.Stream.Close(); err != nil {
+		return nil, err
+	}
+	_result := cs.svcInfo.MethodInfo(cs.method).NewResult().(*generic.Result)
+	if err = cs.Stream.RecvMsg(_result); err != nil {
+		return nil, err
+	}
+	return _result.GetSuccess(), nil
+}
+
+type serverStreamingClient struct {
+	streaming.Stream
+	svcInfo *serviceinfo.ServiceInfo
+	method  string
+}
+
+func NewServerStreaming(ctx context.Context, genericCli Client, method string, req interface{}, callOpts ...callopt.Option) (ServerStreaming, error) {
+	gCli, ok := genericCli.(*genericServiceClient)
+	if !ok {
+		return nil, errors.New("invalid generic client")
+	}
+	stream, err := getStream(ctx, genericCli, method, callOpts...)
+	if err != nil {
+		return nil, err
+	}
+	ss := &serverStreamingClient{Stream: stream, svcInfo: gCli.svcInfo, method: method}
+	_args := gCli.svcInfo.MethodInfo(method).NewArgs().(*generic.Args)
+	_args.Method = method
+	_args.Request = req
+	if err = ss.Stream.SendMsg(_args); err != nil {
+		return nil, err
+	}
+	if err = ss.Stream.Close(); err != nil {
+		return nil, err
+	}
+	return ss, nil
+}
+
+func (ss *serverStreamingClient) Recv() (resp interface{}, err error) {
+	_result := ss.svcInfo.MethodInfo(ss.method).NewResult().(*generic.Result)
+	if err = ss.Stream.RecvMsg(_result); err != nil {
+		return nil, err
+	}
+	return _result.GetSuccess(), nil
+}
+
+type bidirectionalStreamingClient struct {
+	streaming.Stream
+	svcInfo *serviceinfo.ServiceInfo
+	method  string
+}
+
+func NewBidirectionalStreaming(ctx context.Context, genericCli Client, method string, callOpts ...callopt.Option) (BidirectionalStreaming, error) {
+	gCli, ok := genericCli.(*genericServiceClient)
+	if !ok {
+		return nil, errors.New("invalid generic client")
+	}
+	stream, err := getStream(ctx, genericCli, method, callOpts...)
+	if err != nil {
+		return nil, err
+	}
+	return &bidirectionalStreamingClient{Stream: stream, svcInfo: gCli.svcInfo, method: method}, nil
+}
+
+func (bs *bidirectionalStreamingClient) Send(req interface{}) error {
+	_args := bs.svcInfo.MethodInfo(bs.method).NewArgs().(*generic.Args)
```

Comment:
since all neccessary data are solidated, can we just move this logic into `NewBidirectionalStreaming ` too?

Reply:
@AsterDY This creates a new args pointer, so I think we need to create one for each request.